src/SciAssist/pipelines/dataset_extraction.py

```diff
@@ -0,0 +1,307 @@
+from SciAssist.utils.windows_pdf2text import windows_get_bodytext
+
+
+class DatasetExtraction(Pipeline):
+    """
+
+    The pipeline for dataset extraction.
+
+    Args:
+        model_name (`str`, *optional*):
+            A string, the *model name* of a pretrained model provided for this task.
+
+        device (`str`, *optional*):
+            A string, `cpu` or `gpu`.
+
+        cache_dir (`str` or `os.PathLike`, *optional*):
+            Path to a directory in which a downloaded pretrained model should be
+            cached if the standard cache should not be used.
+
+        output_dir (`str` or `os.PathLike`, *optional*):
+            Path to a directory in which the predicted results files should be stored.
+
+        temp_dir (`str` or `os.PathLike`, *optional*):
+            Path to a directory which holds temporary files such as `.tei.xml`.
+
+        tokenizer (PreTrainedTokenizer, *optional*):
+            A specific tokenizer.
+
+        checkpoint (`str` or `os.PathLike`, *optional*):
+            A checkpoint for the tokenizer. You can also specify the `checkpoint` while
+            using the default tokenizer.
+            Can be either:
+                - A string, the *model id* of a predefined tokenizer hosted inside a model repo on huggingface.co.
+                  Valid model ids can be located at the root-level, like `bert-base-uncased`, or namespaced under a
+                  user or organization name, like `allenai/scibert_scivocab_uncased`.
+                - A path to a *directory* containing vocabulary files required by the tokenizer, for instance saved
+                  using the [`~PreTrainedTokenizer.save_pretrained`] method, e.g., `./my_model_directory/`.
+                - A path or url to a single saved vocabulary file if and only if the tokenizer only requires a
+                  single vocabulary file (like Bert or XLNet), e.g.: `./my_model_directory/vocab.txt`. (Not
+                  applicable to all derived classes)
+
+        model_max_length (`int`, *optional*): The max sequence length the model accepts.
+
+    """
+
+    def __init__(
+            self, model_name: Optional[str] = "default", device: Optional[str] = "gpu",
+            cache_dir = None,
+            output_dir = None,
+            temp_dir = None,
+            tokenizer: PreTrainedTokenizer = None,
+            checkpoint = "roberta-base",
+            model_max_length = 128,
+            os_name = None,
+    ):
+
+        super().__init__(task_name = "dataset-extraction", model_name = model_name, device = device,
+                         cache_dir = cache_dir, output_dir = output_dir, temp_dir = temp_dir)
+
+        self.data_utils = self.data_utils(
+            tokenizer = tokenizer,
+            checkpoint = checkpoint,
+            model_max_length = model_max_length
+        )
+        self.os_name = os_name if os_name != None else os.name
+
+
+    def extract(
```

Comment:
Do we need to keep the function names consistent？Because the function will be invoked by users. Or  we may use  `__call__`.